frontend/src/components/worksheets/NewRun/NewRun.jsx

```diff
@@ -0,0 +1,475 @@
+                {dependencies.map((dep, idx) => (
+                    <Grid item container direction="row" key={idx}>
+                        <Grid item xs={4}>
+                            <Typography variant="body1">
+                                {`${dep.target.name} (${shorten_uuid(dep.target.uuid)})`}
+                            </Typography>
+                        </Grid>
+                        <Grid item xs={1} container justify="center">
+                            <Typography variant="body2">as</Typography>
+                        </Grid>
+                        <Grid item xs={3}>
+                            <ConfigTextInput
+                                value={dep.alias}
+                                onValueChange={(alias) => updateDependency(idx, alias)}
+                            />
+                        </Grid>
+                        <Grid item xs={1} container justify="center">
+                            <IconButton
+                                onClick={() => removeDependency(idx)}>
+                                <DeleteIcon fontSize="small" />
+                            </IconButton>
+                        </Grid>
+                    </Grid>
+                ))}
+
+                {/* New dependency ------------------------------------------------------------- */}
+                <Grid item container direction="row" key={-1}>
+                    <Grid item xs={4}>
+                        <Select
+                            options={candidates.map((bundle) => ({
+                                label: `${bundle.name} (${shorten_uuid(bundle.uuid)})`,
+                                value: bundle,
+                            }))}
+                            value=""
+                            onChange={(option) => addDependency(option.value)}
+                            placeholder="target"
+                            noOptionsMessage={() => "No matching bundles"}
+                            components={{
+                                IndicatorsContainer: (props) => null,
+                            }}
+                            styles={{
+                                control: (provided) => ({
+                                    ...provided,
+                                    minHeight: 30,
+                                }),
+                            }}
+                        />
+                    </Grid>
+                    <Grid item xs={1} container justify="center">
+                        <Typography variant="body2">as</Typography>
+                    </Grid>
+                    <Grid item xs={3}>
+                        <ConfigTextInput disabled value="alias"/>
+                    </Grid>
+                </Grid>
+            </Grid>
+        );
+    }
+}
+const DependencyEditor = withStyles((theme) => ({
+    container: {
+        paddingBottom: theme.spacing.large,
+    }
+}))(DependencyEditorRaw);

```

Comment:
Use same convention all around (I also saw `...Base` instead of `...Raw`).  But whatever is fine.